modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/FileSpec.kt

```diff
@@ -3,36 +3,96 @@ package com.beust.kobalt
 import com.beust.kobalt.misc.log
 import java.io.File
 import java.nio.file.*
+import java.nio.file.FileVisitResult.CONTINUE
 import java.nio.file.attribute.BasicFileAttributes
+import java.util.*
+import kotlin.collections.*
 
 sealed class IFileSpec {
     abstract fun toFiles(directory: String): List<File>
+    abstract fun toFile(file: String): List<File>
 
     class FileSpec(val spec: String) : IFileSpec() {
         override public fun toFiles(directory: String) = listOf(File(spec))
+        override public fun toFile(file: String) = listOf(File(spec))
 
         override public fun toString() = spec
     }
 
-    class GlobSpec(val spec: String) : IFileSpec() {
+    class GlobSpec(val includeSpec: ArrayList<String>, val excludeSpec: ArrayList<String>) : IFileSpec() {
+
+        constructor(spec: String) : this(arrayListOf(spec), arrayListOf())
+
         override public fun toFiles(directory: String): List<File> {
+
             val result = arrayListOf<File>()
+            val includeMatchers = prepareMatchers(includeSpec.toTypedArray())
+            val excludeMatchers = prepareMatchers(excludeSpec.toTypedArray())
 
-            val matcher = FileSystems.getDefault().getPathMatcher("glob:$spec")
             Files.walkFileTree(Paths.get(directory), object : SimpleFileVisitor<Path>() {
                 override public fun visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult {
+
                     val rel = Paths.get(directory).relativize(path)
-                    if (matcher.matches(rel)) {
-                        log(3, "Adding ${rel.toFile()}")
-                        result.add(rel.toFile())
+                    excludeMatchers.forEach {
+                        if (it.matches(rel)) {
+                            log(3, "Removing ${rel.toFile()}")
+                            return CONTINUE
+                        }
+                    }
+                    includeMatchers.forEach {
+                        if (it.matches(rel)) {
+                            log(3, "Adding ${rel.toFile()}")
+                            result.add(rel.toFile())
+                            return CONTINUE
+                        }
                     }
-                    return FileVisitResult.CONTINUE
+
+                    return CONTINUE
                 }
             })
             return result
         }
 
-        override public fun toString() = spec
+        override public fun toFile(file: String): List<File> {
+
+            val result = arrayListOf<File>()
+            val includeMatchers = prepareMatchers(includeSpec.toTypedArray())
+            val excludeMatchers = prepareMatchers(excludeSpec.toTypedArray())
+
+            val rel = Paths.get(file)
+            excludeMatchers.forEach {
```

Comment:
BIt of duplication with the code above, we can probably extract some common code here (doesn't need to be in this PR).